clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -305,13 +312,28 @@ public void completed(
           String message,
           Optional<LeaderFollowerStateType> leaderState) {
         // Use thread pool to handle the completion reporting to make sure it is not blocking the report.
-        if (isTopicPartitionIngesting(kafkaTopic, partition)) {
+        if (isTopicPartitionHosted(kafkaTopic, partition)) {
           getCompletionHandlingExecutor().submit(() -> {
             VeniceStoreVersionConfig config = getConfigLoader().getStoreConfig(kafkaTopic);
             config.setRestoreDataPartitions(false);
             config.setRestoreMetadataPartition(false);
             // Start partition consumption locally.
-            startConsumption(config, partition, leaderState);
+            try {
+              startConsumptionLocally(config, partition, leaderState);
```

Comment:
Can you add a comment/javadoc to `startConsumption` to denote that it is for consumption in the isolated process?

Actually, it seems my understanding is wrong and `startConsumption` can handle both cases. Can you please clarify this?

Reply:
Yeah, IsolatedIngestionBackend.startConsumption will start consumption based on resource location (if not exist will by default start in child process)
startConsumptionLocally is to enforce it to start locally as here is the handover logic from child -> main

Reply:
I can add more Java docs